### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by jjonhwa** [ultralytics/yolov5#6465]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

All conditions were entered the same and the following seeds were used to proceed. However, the resulting value is output a little differently. In other words, it is not reproducible, so may I know where this problem occurs? It's hard to know where it comes from even though I checked the code due to a lack of my skills. LOL

I am using colab pro and seed was entered as follows.
![image](https://user-images.githubusercontent.com/53552847/151630732-809f02c9-f6e6-423f-9d19-ff407024c8d8.png)
![image](https://user-images.githubusercontent.com/53552847/151630820-a475b62a-0cad-4bc6-bd16-baafc4e09fb3.png)

The output of my wandb is as follows. The conditions of both experiments are the same. Loss is output almost the same. However, the output of precision and recall show quite different patterns.
![image](https://user-images.githubusercontent.com/53552847/151631101-5678065b-02b6-409e-9d77-72614fb70625.png)
![image](https://user-images.githubusercontent.com/53552847/151631125-5fd6a2fc-b72a-4f3b-8640-69b21e92df78.png)

**Comment by glenn-jocher** [comment]
@jjonhwa 👋 Hello! Thanks for asking about **training reproducibility**. YOLOv5 🚀 uses a single training seed which is set here using the `init_seeds()` function.
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/train.py#L108

This function sets `python`, `numpy` and `torch` seeds and updates `cudnn` settings:
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/utils/general.py#L122-L130

To set a new seed for example:
```python
    init_seeds(2 + RANK)  # set seed=2
```

Note that even when using the same seed trainings may produce different results, especially when using CUDA backends. See https://pytorch.org/docs/stable/notes/randomness.html for details on factors affecting PyTorch training reproducibility and sources of randomness.

<img width="949" alt="Screen Shot 2022-01-08 at 11 20 59 AM" src="https://user-images.githubusercontent.com/26833433/148660366-9155dfd9-351c-4174-818f-2b1b907b3e4e.png">

Good luck 🍀 and let us know if you have any other questions!

**Comment by jjonhwa** [comment]
Oh, I understand. Thank you for helping me.
